### DISCUSSION TITLE
Help with batch adjust 100's of image exposures based on a reference image/histogram

### DISCUSSION THREAD

**Discussion by imagecatcher** [ImageMagick/ImageMagick#3291] [Digital Image Processing]
With aerial/drone surveying I generate 100's of images traversing the terrain in the day. Ambient light is dynamically changing. The camera is generally in shutter priority. Nevertheless, when images are stitched (photogrammetry) often there is banding/exposure differences between traverses. 
I have never coded (well 30yrs ago with basic and cobalt). I suspect I need to identify a 'reference' image that is well exposed and colour balanced OR adjust one in Lightroom. All other images will compare themselves to the histogram to match/approximate it within a ISO limit to limit noise. Each image will need to apply a dynamic approach across exposure, white/black point, the color space and intensity(in video talk, the ire). All while preserving the original images therefore creating a new folder with the adjusted imgs. All other exif metadata such as gps and time needs to stay intact.
The attached image below is after applying Lightroom adjustments to small similar groups /runs of images in. But until it's stitched (the last step) its pot luck. 

I'm hoping existing modules of code can be fashioned together. (@snibgo) seems to be all over this area. I use both mac & pc with no preference on which sandpit I play in.
Thanks out there.

![IMG_1138 2](https://user-images.githubusercontent.com/1073921/108979981-5291bf00-76df-11eb-91d2-5a95d51883e5.JPG)

**Comment by snibgo** [comment]
IM contains low-level operations for doing this work. But a complete system for doing this would be complex: organizing the images, figuring which images overlap with others and what to do with that overlap, processing changes in terrain (eg fields with crops, fields that are bare earth), changes of lighting within a single day of images or across multiple days, scaling according to height of camera and topography, camera calibration, cloud processing, and so on.

If you have limited experience of image processing or coding or scripting, this would be a daunting project.

With recent popularity of drones, applications have been developed for this purpose. Some are free; some are not. For example: https://www.opendronemap.org/ and https://dronemapper.com/ . I have no experience with any of these.

**Comment by imagecatcher** [comment]
Hi there Snibgo,
Thanks for chiming in. I think by adding the photo I may have mislead you. I am not after a Photogrammetry solution as I already have that with Pix4dMapper software. And the picture in the commit was the output of a recent job processed with Pix4d. Pix does do a little ‘blend’ but not enough. What I am trying to achieve is to ‘prep’ a folder of the images before the photogrammetry mapping software process begins. Really basic photo editing adjustments but around a reference image and as an automated batch. (not basic coding for me however ;)
Currently I use Lightroom. It might have a 1000 images, I find a reference one, find a string of crappy ones and apply a Fixed set of adjustments to bring it close to the reference img, Repeat for all the similar strings of crappy shots. Lr can do “Match Total Exposure” which does an ok job but it is strictly exposure. It does not consider any other aspect like shadow, white point etc.
As I write, I have had an idea. I do have LrTimelapse software which does change images dynamically around keyframes… this could be the trick and re-purposed.
Thanks for looking in on this. I will update my research.

**Comment by snibgo** [comment]
Many methods can be used to adjust one image to make it look like another. For example, in rough order of complexity and precision:

[Setting the mean](http://im.snibgo.com/setmean.htm)
[Gain and bias](http://im.snibgo.com/gainbias.htm)
[Match histograms](http://im.snibgo.com/customim.htm)
[Sparse hald cluts](http://im.snibgo.com/sphaldcl.htm)
[Colours to matrix and polynomials](http://im.snibgo.com/col2mp.htm)
